statsprinter.go

```diff
@@ -159,44 +212,80 @@ func (p *planePrinter) printStatistics(t tcping) {
 }
 
 func (p *planePrinter) printProbeSuccess(hostname, ip string, port uint16, streak uint, rtt float32) {
+	var message string
 	if hostname == "" {
-		colorLightGreen("Reply from %s on port %d TCP_conn=%d time=%.3f ms\n",
+		message = fmt.Sprintf("Reply from %s on port %d TCP_conn=%d time=%.3f ms\n",
 			ip, port, streak, rtt)
-		return
+	} else {
+		message = fmt.Sprintf("Reply from %s (%s) on port %d TCP_conn=%d time=%.3f ms\n",
+			hostname, ip, port, streak, rtt)
 	}
-
-	colorLightGreen("Reply from %s (%s) on port %d TCP_conn=%d time=%.3f ms\n",
-		hostname, ip, port, streak, rtt)
+	options := &PrintOptions{
+		color:         LightGreen,
+		message:       message,
+		showTimestamp: p.showTimestamp,
+	}
+	printReply(options)
 }
 
 func (p *planePrinter) printProbeFail(hostname, ip string, port uint16, streak uint) {
+	var message string
 	if hostname == "" {
-		colorRed("No reply from %s on port %d TCP_conn=%d\n",
-			ip, port, streak)
-		return
+		message = fmt.Sprintf("No reply from %s on port %d TCP_conn=%d\n", ip, port, streak)
+	} else {
+		message = fmt.Sprintf("No reply from %s (%s) on port %d TCP_conn=%d\n", hostname, ip, port, streak)
+	}
+	options := &PrintOptions{
+		color:         Red,
+		message:       message,
+		showTimestamp: p.showTimestamp,
 	}
-	colorRed("No reply from %s (%s) on port %d TCP_conn=%d\n",
-		hostname, ip, port, streak)
+	printReply(options)
 }
 
 func (p *planePrinter) printTotalDownTime(downtime time.Duration) {
-	colorYellow("No response received for %s\n", durationToString(downtime))
+	message := fmt.Sprintf("No response received for %s\n", durationToString(downtime))
+	options := &PrintOptions{
+		color:   Yellow,
+		message: message,
+	}
+	printReply(options)
 }
 
 func (p *planePrinter) printRetryingToResolve(hostname string) {
-	colorLightYellow("retrying to resolve %s\n", hostname)
+	message := fmt.Sprintf("retrying to resolve %s\n", hostname)
+	options := &PrintOptions{
+		color:   LightYellow,
+		message: message,
+	}
+	printReply(options)
 }
 
 func (p *planePrinter) printInfo(format string, args ...any) {
-	colorLightBlue(format+"\n", args...)
+	message := fmt.Sprintf(format+"\n", args...)
+	options := &PrintOptions{
+		color:   LightCyan,
+		message: message,
+	}
+	printReply(options)
 }
 
 func (p *planePrinter) printError(format string, args ...any) {
-	colorRed(format+"\n", args...)
+	message := fmt.Sprintf(format+"\n", args...)
+	options := &PrintOptions{
+		color:   Red,
+		message: message,
+	}
+	printReply(options)
 }
 
 func (p *planePrinter) printVersion() {
-	colorGreen("TCPING version %s\n", version)
+	message := fmt.Sprintf("TCPING version %s\n", version)
+	options := &PrintOptions{
+		color:   LightYellow,
+		message: message,
+	}
+	printReply(options)
```

Comment:
these are still here? Similar to the rest

Reply:
I think you meant that it was carrying the wrong color?
Yeah I fixed that.